templates/virt-install-script.sh.j2

```diff
@@ -1,29 +1,30 @@
-#!/bin/bash
-
-if [ ! -f /etc/libvirt/qemu/{{ inventory_hostname }}.xml ]; then
-
 virt-install \
   --name={{ inventory_hostname }} \
-  --os-variant=rhel6 \
-{% if virt_install_cpu_model %}
-  --cpu={{ virt_install_cpu_model }} \
-{% else %}
-  --cpu=host \
-{% endif %}
   --vcpus={{ vcpus }} \
   --ram={{ ram }} \
 {% for disk in disks %}
-  --disk path={{ disk.path }},cache=writethrough,device=disk,bus=virtio,size={{ disk.size }},format=raw \
+  --disk path={{ disk.path }},cache=writethrough,device=disk,bus=virtio,size={{ disk.size }},format={{ disk.format | default('raw') }} \
 {% endfor %}
 {% for bridge in bridges %}
   --network bridge={{ bridge }},model=virtio \
 {% endfor %}
   --connect=qemu:///system \
-  --location={{ install_url }} \
-  --graphics=vnc,keymap="fi" \
-  --noautoconsole \
   --autostart \
+  --noautoconsole \
+  --events on_poweroff=preserve \
```

Comment:
I tested kickstart deployments today. Setting `on_poweroff=preserve` for the domain breaks them. It will cause the installation to go into a loop where the install starts over and over again and never finishes.

Perhaps the problem is that when you do image based deployments the domain doesn't get defined properly and thus it would be an issue if `on_poweroff` is set to `destroy`? I tried to verify if this is the case, but I ran into another issue:

```
FAILED! => {"changed": false, "failed": true, "msg": "No package matching 'libguestfs-tools' found available, installed or updated", "rc": 126, "results": ["No package matching 'libguestfs-tools' found available, installed or updated"]}
```

This seems to be because `become: yes` is not set for the task used for installing libguestfs-tools. Perhaps you are using this role directly as root?